compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/PsiUtils.kt

```diff
@@ -91,62 +92,85 @@ public fun KtAnnotated.requireAnnotation(
 }
 
 @ExperimentalAnvilApi
+public fun KtAnnotated.requireAnnotations(
+  fqName: FqName,
+  module: ModuleDescriptor
+): List<KtAnnotationEntry> {
+  return findAnnotations(fqName, module)
+    .ifEmpty { null }
+    ?: throw AnvilCompilationException(
+      element = this,
+      message = "Couldn't find the annotation $fqName."
+    )
+}
+
+@ExperimentalAnvilApi
+@Deprecated("Repeatable")
 public fun KtAnnotated.findAnnotation(
   fqName: FqName,
   module: ModuleDescriptor
-): KtAnnotationEntry? {
+): KtAnnotationEntry? = findAnnotations(fqName, module).singleOrEmpty()
+
+@ExperimentalAnvilApi
+public fun KtAnnotated.findAnnotations(
+  fqName: FqName,
+  module: ModuleDescriptor
+): List<KtAnnotationEntry> {
   val annotationEntries = annotationEntries
-  if (annotationEntries.isEmpty()) return null
+  if (annotationEntries.isEmpty()) return emptyList()
```

Comment:
Similar comment, but the logic below will return an emptyList() automatically if it's empty, so why do the initial check first? It's my understanding that the collection extensions have early bail out mechanisms baked in. i.e. filter {}

Reply:
The operations below are quite heavy and returning early here happens quite often. It's a quick, faster exit.